### DISCUSSION TITLE
What todo with warning - control reaches end of non-void function

### DISCUSSION THREAD

**Discussion by SeleznevPavel** [citusdata/citus#7678] [General]
At the following line
https://github.com/citusdata/citus/blob/main/src/backend/columnar/columnar_tableam.c#L3022

I get warning that prevent build 
"columnar_tableam.c:3024:1: error: control reaches end of non-void function"

Treat warnings as error is turned on for the whole project

What is the best way to get around this issue?

Option 1 add fake return
return null

Option 2 put the function into 
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wreturn-type"
static char *
AvailableExtensionVersionColumnar(void)
{
....
}
#pragma GCC diagnostic pop

I am planning create pull request with best solition

**Comment by EvgenyNechayev** [comment]
Hi, Pavel! IMHO, the first solution is preferred, as it is a commonly used approach in the PostgreSQL codebase.
return NULL;	/* keep compiler quiet */

**Comment by SeleznevPavel** [comment]
https://github.com/citusdata/citus/pull/7680
